File: refactor.js
const fs = require('fs');

let code = fs.readFileSync('server.js', 'utf8');

const funcs_to_async = [
    'getLimiteSentados', 'getBancosTraseiros', 'calcularAssentosDoDia',
    'calcularAssentosIdaVolta', 'calcularAssentosTrechoComPonteiro',
    'calcularBancosTraseirosComExclusao', 'calcularBancosTraseiros',
    'avancarPonteiroAssentos', 'avancarPonteiroBancos',
    'gerarRelatorioDoDia', 'limparRespostasDia'
];

funcs_to_async.forEach(fn => {
    // Change function declarations to async
    const regexFunc = new RegExp(`function ${fn}\\(`, 'g');
    code = code.replace(regexFunc, `async function ${fn}(`);

    // Change function calls to await (only if not preceded by async function)
    const regexCall = new RegExp(`(?<!async function )${fn}\\(`, 'g');
    code = code.replace(regexCall, `await ${fn}(`);
});

// Make Express routes async
code = code.replace(/app\.(get|post|put|delete)\(([^,]+),\s*(limiteGeral|limiteEscrita),\s*\(req, res\) => {/g, 'app.$1($2, $3, async (req, res) => {');
code = code.replace(/app\.(get|post|put|delete)\(([^,]+),\s*\(req, res\) => {/g, 'app.$1($2, async (req, res) => {');

// Replace db.prepare(SQL).get(args)
code = code.replace(/db\.prepare\((.*?)\)\.get\((.*?)\)/g, (match, sql, args) => {
    if (args.trim()) {
        return `(await db.execute({ sql: ${sql}, args: [${args}] })).rows[0]`;
    }
    return `(await db.execute(${sql})).rows[0]`;
});

// Replace db.prepare(SQL).all(args)
code = code.replace(/db\.prepare\((.*?)\)\.all\((.*?)\)/g, (match, sql, args) => {
    if (args.trim()) {
        return `(await db.execute({ sql: ${sql}, args: [${args}] })).rows`;
    }
    return `(await db.execute(${sql})).rows`;
});

// Replace db.prepare(SQL).run(args)
code = code.replace(/db\.prepare\((.*?)\)\.run\((.*?)\)/g, (match, sql, args) => {
    if (args.trim()) {
        return `await db.execute({ sql: ${sql}, args: [${args}] })`;
    }
    return `await db.execute(${sql})`;
});

// Fix cron jobs to have async functions if they use await inside
code = code.replace(/cron\.schedule\('(.*?)', \(\) => {/g, "cron.schedule('$1', async () => {");

fs.writeFileSync('server_refactored.js', code, 'utf8');
console.log('Refactoring complete!');

File: refactor.py
import re

def refactor():
    with open('server.js', 'r', encoding='utf8') as f:
        code = f.content = f.read()

    # Change function definitions to async
    funcs_to_async = [
        'getLimiteSentados', 'getBancosTraseiros', 'calcularAssentosDoDia',
        'calcularAssentosIdaVolta', 'calcularAssentosTrechoComPonteiro',
        'calcularBancosTraseirosComExclusao', 'calcularBancosTraseiros',
        'avancarPonteiroAssentos', 'avancarPonteiroBancos',
        'gerarRelatorioDoDia', 'limparRespostasDia'
    ]
    
    for fn in funcs_to_async:
        code = re.sub(fr'function {fn}\(', f'async function {fn}(', code)

    # Change function calls to await
    for fn in funcs_to_async:
        code = re.sub(fr'(?<!async function ){fn}\(', f'await {fn}(', code)

    # Make Express routes async
    code = re.sub(r'app\.(get|post|put|delete)\(([^,]+),\s*(limiteGeral|limiteEscrita),\s*\(req, res\) => {', r'app.\1(\2, \3, async (req, res) => {', code)
    code = re.sub(r'app\.(get|post|put|delete)\(([^,]+),\s*\(req, res\) => {', r'app.\1(\2, async (req, res) => {', code)

    # db.prepare('SELECT ...').get(args) -> (await db.execute({ sql: 'SELECT ...', args: [args] })).rows[0]
    code = re.sub(
        r'db\.prepare\((.*?)\)\.get\((.*?)\)',
        lambda m: f"(await db.execute({{ sql: {m.group(1)}, args: [{m.group(2)}] }})).rows[0]" if m.group(2).strip() else f"(await db.execute({m.group(1)})).rows[0]",
        code
    )

    # db.prepare('SELECT ...').all(args) -> (await db.execute({ sql: 'SELECT ...', args: [args] })).rows
    code = re.sub(
        r'db\.prepare\((.*?)\)\.all\((.*?)\)',
        lambda m: f"(await db.execute({{ sql: {m.group(1)}, args: [{m.group(2)}] }})).rows" if m.group(2).strip() else f"(await db.execute({m.group(1)})).rows",
        code
    )

    # db.prepare('INSERT...').run(args) -> await db.execute({ sql: 'INSERT...', args: [args] })
    code = re.sub(
        r'db\.prepare\((.*?)\)\.run\((.*?)\)',
        lambda m: f"await db.execute({{ sql: {m.group(1)}, args: [{m.group(2)}] }})" if m.group(2).strip() else f"await db.execute({m.group(1)})",
        code
    )

    with open('server_refactored.js', 'w', encoding='utf8') as f:
        f.write(code)

if __name__ == '__main__':
    refactor()
